lib/modules/dosomething/dosomething_northstar/dosomething_northstar.auth.inc

```diff
@@ -135,15 +142,17 @@ function dosomething_northstar_openid_connect_post_authorize($tokens, $account,
  * Save a post-authorization destination in the session (e.g. for signing
  * up for a campaign).
  *
- * @param $action
- * @param $nid
+ * @param string $action
+ * @param string $nid
+ * @param array $param
  */
-function dosomething_northstar_set_authorization_action($action, $nid) {
+function dosomething_northstar_set_authorization_action($action, $nid, $params = []) {
   if ($action !== 'signup') {
     return;
   }
 
   $_SESSION['openid_connect_action'] = ['signup' => $nid];
+  $_SESSION['affiliate_messaging_opt_in'] = isset($params['affiliate_messaging_opt_in']) ? $params['affiliate_messaging_opt_in'] : FALSE;
```

Comment:
I think this makes sense to minimize the number of unique keys we store on the session and include as part of the existing `signup` post-auth action:

```php
$_SESSION['openid_connect_action'] = [
    'signup' => [
        'nid' => $nid,
        'params' => [
                'messaging_opt_in' => ! empty($params['affiliate_messaging_opt_in'])
        ],
    ]
];
```

Then we can grab all of the details of the signup from one place:

```php
 function dosomething_northstar_perform_authorization_actions() {
    global $user;
 
    if (! isset($_SESSION['openid_connect_action'])) {
     return;
    }
  
    $actions = $_SESSION['openid_connect_action'];

    foreach ($actions as $action => $target) {
        if ($action === 'signup') {
            $source = dosomething_signup_get_query_source();
            dosomething_signup_user_signup((int) $target['nid'], $user, $source, $target['params']);
        }
    }
}
```